awx/main/models/events.py

```diff
@@ -586,6 +604,23 @@ def _update_host_summary_from_stats(self, hostnames):
     def job_verbosity(self):
         return self.job.verbosity
 
+    # If the Django primary key and the partition key have values, try update_or_create()
+    # Otherwise, fallback to inherited save()
+    def save(self, *args, **kwargs):
```

Comment:
I think it might be worth pulling the changes in this file out into a separate PR while we think long and hard about the migration you added.

Reply:
The JobEvent model has been restored.